lib/jenkins_pipeline/job.rb

```diff
@@ -27,10 +30,11 @@ def serialize
 
     private
 
-    def build_status(last_build, result)
-      return "running" if build_running?(last_build)
+    def build_status
+      return "undefined" if %w(running undefined failure).include?(@upstream_status)
+      return "running" if build_running?(@last_build)
       results = { success: "success", failure: "failure" }
-      results[result.to_sym]
+      results[@result.to_sym]
```

Comment:
Do we need this variable `results` ?  The keys are pretty much the same the values
Instead of that, we could only return `@result`